homeassistant/components/switch/xiaomi_miio.py

```diff
@@ -38,31 +38,53 @@
 @asyncio.coroutine
 def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
     """Set up the switch from config."""
-    from mirobo import Plug, DeviceException
+    from mirobo import Device, DeviceException
 
     host = config.get(CONF_HOST)
     name = config.get(CONF_NAME)
     token = config.get(CONF_TOKEN)
 
     _LOGGER.info("Initializing with host %s (token %s...)", host, token[:5])
 
+    devices = []
     try:
-        plug = Plug(host, token)
+        plug = Device(host, token)
         device_info = plug.info()
         _LOGGER.info("%s %s %s initialized",
                      device_info.raw['model'],
                      device_info.raw['fw_ver'],
                      device_info.raw['hw_ver'])
 
-        xiaomi_plug_switch = XiaomiPlugSwitch(name, plug, device_info)
+        if device_info.raw['model'] in ['chuangmi.plug.v1']:
+            from mirobo import PlugV1
+            plug = PlugV1(host, token)
+
+            for channel_usb in [True, False]:
+                device = ChuangMiPlugV1Switch(
+                    name, plug, device_info, channel_usb)
+                devices.append(device)
+
+        elif device_info.raw['model'] in ['qmi.powerstrip.v1',
+                                          'zimi.powerstrip.v2']:
+            from mirobo import Strip
+            plug = Strip(host, token)
+            device = XiaomiPowerStripSwitch(name, plug, device_info)
+            devices.append(device)
+        else:
+            # chuangmi.plug.m1, chuangmi.plug.v2
```

Comment:
Maybe check also with elif, and error out on an unknown device to let the users to report back on such cases?